core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -329,6 +329,18 @@ sealed trait ZIO[-R, +E, +A]
   ): ZIO[R1, E1, A1] =
     catchSomeDefect { case t => h(t) }
 
+  /**
+   * Recovers from all failures with provided function.
+   *
+   * {{{
+   * effect.catchAllFailure(_ => backup())
+   * }}}
```

Comment:
This is going to be very confusing to users on why to use this method over `catchAllZIO`. We need to provide a better example in the scaladoc that differentiates, but also point to `catchAllZIO` when the users don't care about retrieving the full Cause of the failure (which is in most cases in user code)

Reply:
You meant `catchAllCause` or `catchAll`?

Reply:
documentation improved.